File: backend/src/Web/Endpoints/Expenses.cs
using CleanArchitecture.Application.Expenses.Commands.CreateExpense;
using CleanArchitecture.Application.Expenses.Queries;
using CleanArchitecture.Application.Expenses.Queries.GetExpenseById;
using CleanArchitecture.Application.Expenses.Queries.GetExpenses;
using Microsoft.AspNetCore.Http.HttpResults;

namespace CleanArchitecture.Web.Endpoints;

public class Expenses : EndpointGroupBase
{
    public override string? GroupName => "expenses";

    public override void Map(RouteGroupBuilder group)
    {
        group.RequireAuthorization();

        group.MapGet("", GetExpenses);
        group.MapGet("{id}", GetExpenseById);
        group.MapPost(CreateExpense);
    }

    public async Task<Ok<List<ExpenseDto>>> GetExpenses(ISender sender, [AsParameters] GetExpensesQuery query)
    {
        return TypedResults.Ok(await sender.Send(query));
    }

    public async Task<Results<Ok<ExpenseDto>, NotFound>> GetExpenseById(ISender sender, int id)
    {
        var item = await sender.Send(new GetExpenseByIdQuery(id));
        return TypedResults.Ok(item);
    }

    public async Task<Created<ExpenseDto>> CreateExpense(ISender sender, CreateExpenseCommand command)
    {
        var result = await sender.Send(command);
        return TypedResults.Created($"/api/expenses/{result.Id}", result);
    }
}

File: front/hotel-frontend/src/api/expenses.ts
import http from './http';
import type { CreateExpenseCommand, ExpenseDto, GetExpensesParams, ExpensesSummaryDto } from './types/expenses';

export const getExpenses = async (params: GetExpensesParams): Promise<ExpensesSummaryDto> => {
    const response = await http.get<ExpensesSummaryDto>('expenses', { params });
    return response.data;
};

export const getExpenseById = async (id: number): Promise<ExpenseDto> => {
    const response = await http.get<ExpenseDto>(`expenses/${id}`);
    return response.data;
};

export const createExpense = async (command: CreateExpenseCommand): Promise<ExpenseDto> => {
    const response = await http.post<ExpenseDto>('expenses', command);
    return response.data;
};
